lib/pbench/agent/tool_data_sink.py

```diff
@@ -414,10 +476,21 @@ def _cleanup(self):
         """_cleanup - Encapsulates the proper shutdown sequence for the WSGI server
         and Redis Server connection.
         """
-        self.logger.debug("unsubscribe")
-        self._pubsub.unsubscribe()
-        self.logger.debug("pubsub close")
-        self._pubsub.close()
+        if self._pubsub:
+            self.logger.debug("unsubscribe client")
+            self._pubsub.unsubscribe()
+            self.logger.debug("pubsub close client")
+            self._pubsub.close()
+        try:
+            self.tm_log_capture_thread.join()
```

Comment:
Why join with the logging thread first?  Because we unsubscribe as part of termination, which shutdowns the channel cleanly so that we only take the heavy hammer approach during whacky exception handling.